packages/wallet/src/mock-rollup-state-machine.ts

```diff
@@ -0,0 +1,209 @@
+      return {
+        uni: 0,
+        pigi: 0,
+      }
+    }
+    return this.state[account].balances
+  }
+
+  public getUniswapBalances(): Balances {
+    return this.state[UNISWAP_ADDRESS].balances
+  }
+
+  private ecdsaRecover(signature: MockedSignature): Address {
+    // TODO: Move this out of this class and instead put in keystore
+    return signature
+  }
+
+  public applyTransaction(
+    signedTransaction: SignedTransaction
+  ): TransactionReceipt {
+    const sender: Address = signedTransaction.signature
+    const transaction: Transaction = signedTransaction.transaction
+    if (isTransferTransaction(transaction)) {
+      return this.applyTransfer(sender, transaction)
+    } else if (isSwapTransaction(transaction)) {
+      return this.applySwap(sender, transaction)
+    }
+    throw new InvalidTransactionTypeError()
+  }
+
+  private getTxReceipt(stateUpdate: any): TransactionReceipt {
+    return {
+      aggregatorSignature: 'MOCKED',
+      stateUpdate,
+    }
+  }
+
+  private hasEnoughMoney(
+    account: string,
+    tokenType: TokenType,
+    greaterThanAmount: number
+  ) {
+    // Check that the account has more than some amount of pigi/uni
+    if (
+      !(account in this.state) ||
+      this.state[account].balances[tokenType] < greaterThanAmount
+    ) {
+      return false
+    }
+    return true
+  }
+
+  private applyTransfer(
+    sender: Address,
+    transfer: Transfer
+  ): TransactionReceipt {
+    // Make sure the amount is above zero
+    if (transfer.amount < 1) {
+      throw new NegativeAmountError()
+    }
+    // Check that the sender has enough money
+    if (!this.hasEnoughMoney(sender, transfer.tokenType, transfer.amount)) {
+      throw new InsufficientBalanceError()
+    }
+    // Make sure we've got a record for the recipient
+    if (!(transfer.recipient in this.state)) {
+      // If not, make a record
+      this.state[transfer.recipient] = {
+        balances: {
+          uni: 0,
+          pigi: 0,
+        },
+      }
+    }
+    // Update the balances
+    this.state[sender].balances[transfer.tokenType] -= transfer.amount
+    this.state[transfer.recipient].balances[transfer.tokenType] +=
+      transfer.amount
+    return this.getTxReceipt({
+      sender: this.state[sender],
+      recipient: this.state[transfer.recipient],
+    })
+  }
+
+  private applySwap(sender: Address, swap: Swap): TransactionReceipt {
+    // Make sure the amount is above zero
+    if (swap.inputAmount < 1) {
+      throw new NegativeAmountError()
+    }
+    // Check that the sender has enough money
+    if (!this.hasEnoughMoney(sender, swap.tokenType, swap.inputAmount)) {
+      throw new InsufficientBalanceError()
+    }
+    // Check that we'll have ample time to include the swap
+    // TODO
+
+    // Set the post swap balances
+    ;[
+      this.state[sender].balances,
+      this.state[UNISWAP_ADDRESS].balances,
+    ] = this.getPostSwapBalances(
+      swap,
+      this.state[sender].balances,
```

Comment:
`this.getPostSwapBalances` should have access to `this.state` and `UNISWAP_ADDRESS`, so you should just be able to pass `swap` and `sender` and modify the `this.state` objects in that function instead of returning new objects and overwriting `balances`